01_prepare_host.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+set -eux
+
+PROJECT_DIR=$(dirname -- $(readlink -e -- ${BASH_SOURCE[0]}))
+
+if [[ "$(id -u)" -eq 0 ]]; then
+  echo "Please run as a non-root user"
+  exit 1
+fi
+
+# Check OS type and version
+source /etc/os-release
+export DISTRO="${ID}${VERSION_ID%.*}"
+export OS="${ID}"
+export OS_VERSION_ID="${VERSION_ID}"
+export SUPPORTED_DISTROS=(ubuntu22)
+
+if [[ ! "${SUPPORTED_DISTROS[*]}" =~ ${DISTRO} ]]; then
+  echo "Supported OS distros for the host are: Ubuntu 22.04"
+  exit 1
+fi
+
+
+if [[ "${OS}" = "ubuntu" ]]; then
+  # Set apt retry limit to higher than default to
+  # make the data retrival more reliable
+  sudo sh -c ' echo "Acquire::Retries \"10\";" > /etc/apt/apt.conf.d/80-retries '
+  sudo DEBIAN_FRONTEND=noninteractive apt-get update
+  sudo DEBIAN_FRONTEND=noninteractive apt-get -y install python3-pip python3-dev jq curl wget pkg-config bash-completion
+
+  # Set update-alternatives to python3
+  if [[ "${DISTRO}" = "ubuntu22" ]]; then
+    sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.10 1
+  fi
+fi
+
+sudo python -m pip install ansible
+
+# Install requirements
+ansible-galaxy install -r requirements.yml
+
+# Run ansible prepare host playbook
+export ANSIBLE_ROLES_PATH=$PROJECT_DIR/roles
+ANSIBLE_FORCE_COLOR=true ansible-playbook \
+  -i ${PROJECT_DIR}/inventories/localhost_inventory.yml \
+  $PROJECT_DIR/playbooks/prepare_host.yml $@
+
+# Restart libvirtd service to get the new group membership loaded
+if ! id "${USER}" | grep -q libvirt; then
```

Comment:
I tested the PR on `Ubuntu 22.04`, and it never executes this block here.
This is happening because the installation of `libvirt-daemon-system` automatically adds the current user to the `libvirt` group (but not fully updated and needs logout/login)
```
kris@c3-medium-x86-01:~$ id
uid=1000(kris) gid=1000(kris) groups=1000(kris),27(sudo)
kris@c3-medium-x86-01:~$ id kris
uid=1000(kris) gid=1000(kris) groups=1000(kris),27(sudo),122(libvirt)
``` 
For this reason the code in this block will never be executed and will cause later issues like:
```
"Failed to connect socket to '/var/run/libvirt/libvirt-sock': Permission denied"
```
and

```
Error changing power state at libvirt URI \"qemu:///system\"
```

Reply:
@Kristian-ZH this should now be resolved I believe - I moved the permissions/apparmor workarounds into [roles/libvirt-setup/tasks/libvirt_service_tasks.yml](https://github.com/suse-edge/metal3-demo/pull/34#diff-0b6ea691ca0360cd9d50f433df2799ce60b26c8682d2f5670f5e4687c0544a26) which avoids the script conditional

I tested Ubuntu 22.04 on VM and Equinix this time and seems to be OK, let me know if you can see any remaining issues, thanks!